jetty-maven-plugin/pom.xml

```diff
@@ -128,20 +128,33 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
+      <scope>provided</scope>
       <exclusions>
         <exclusion>
           <groupId>javax.annotation</groupId>
           <artifactId>javax.annotation-api</artifactId>
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-model</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-settings</artifactId>
+      <scope>provided</scope>
+    </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-artifact</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
+      <scope>provided</scope>
```

Comment:
Will have to add an exclusion for `findbugs jsr305`.

Reply:
I read this change as not exposing maven-core / findbugs jsr305 on the plugin classpath.
Perhaps we should add a new `it` test for this, something that can replicate the report in #7677 and then prove out that this PR will fix things?

Reply:
this should not expose transitive dependencies of maven-core because `provided`.
Yup I will have a look to add an IT.